app/src/main/java/org/oppia/android/app/topic/lessons/StorySummaryAdapter.kt

```diff
@@ -148,5 +148,30 @@ class StorySummaryAdapter(
         }
       }
     }
+
+    private fun createRecyclerViewAdapter(
+      storyId: String,
+      chapterList: MutableList<ChapterSummary>
+    ): BindableAdapter<ChapterSummary> {
+      return BindableAdapter.SingleTypeBuilder
+        .newBuilder<ChapterSummary>()
+        .registerViewBinder(
+          inflateView = { parent ->
+            LessonsChapterViewBinding.inflate(
+              LayoutInflater.from(parent.context),
+              parent,
+              /* attachToParent= */ false
+            ).root
+          },
+          bindView = { view, chapterSummary ->
+            val binding = DataBindingUtil.findBinding<LessonsChapterViewBinding>(view)!!
+            binding.chapterSummary = chapterSummary
+            binding.index = chapterList.indexOf(chapterSummary)
+            binding.chapterContainer.setOnClickListener {
+              chapterSummarySelector.selectChapterSummary(storyId, chapterSummary)
+            }
+          }
```

Comment:
Prefer creating a view model class (non-injectable, non-Android view model) that contains the summary, index, and a bound listener callback for chapter selection. Then, you can use that view model as the type instead of ChapterSummary & bind everything through data-binding. That also trivializes bindView & the registration since you can use the databinding inflate/bind functions, instead, such as we do here:

https://github.com/oppia/oppia-android/blob/74db288b9ac25fb942a8feb35e918bcd7e94e55c/app/src/main/java/org/oppia/android/app/help/HelpFragmentPresenter.kt#L46

Reply:
done